### PR TITLE
Add support for shallow link paths

### LINKED ISSUE #3100: file resource link_path always traces link to the target
### 🎛 Description

bug report
could be related to #3099
situation:
```
# ls -l /another_link /link
lrwxrwxrwx. 1 root root  4 May 31 17:47 /another_link -> /usr
lrwxrwxrwx. 1 root root 13 May 31 17:47 /link -> /another_link
```
I need to check that /link points to /another_link and I don't care where /another_link points, but it looks like impossible, because:
```
inspec> file('/link').link_path
=> "/usr"
```

### 🌍 InSpec and Platform Version

inspec --version
2.1.72

### 🤔 Replication Case

see above

### PR DESCRIPTION
**Pull request by ColinHebert**
This should wait on inspec/train#309

Fixes #3100 and #2106

### CODE REVIEW

File: docs/resources/file.md.erb [hunk 1]
```diff
@@ -74,10 +74,17 @@ The following examples show how to use this InSpec audit resource.
...
 
 The `link_path` property tests if the file exists at the specified path. If the file is a symlink,
-InSpec will resolve the symlink and return the ultimate linked file.
+InSpec will resolve the symlink recursively and return the ultimate linked file.
 
     its('link_path') { should eq '/some/path/to/file' }
 
+### shallow_link_path
+
+The `link_path` property tests if the file exists at the specified path. If the file is a symlink,
```
Comment (clintoncwolfe) [inline docs/resources/file.md.erb]: I feel this is an unclear explanation.  A [property](https://www.inspec.io/docs/reference/glossary/#property) doesn't test anything, it returns a value.  Here, you might say, "The `shallow_link_path` property returns the path that the file refers to, only resolving it once (that is, it performs a `readlink` operation).  If the file is not a symlink, `nil` is returned."

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
Just a documentation clarification - otherwise looks great, thanks @ColinHebert !

File: docs/resources/file.md.erb (hunk 1 above)
Comment (ColinHebert) [reply docs/resources/file.md.erb]: Fair, updating to reflect that text

**Review by clintoncwolfe — APPROVED** [review]
Thanks - looks good @ColinHebert!

**Review by jquick — APPROVED** [review]
Thanks @ColinHebert

**Pull request merged**